src/Executor/Executor.php

```diff
@@ -119,11 +126,46 @@ public function createExecution(
         ];
 
         $response = $this->call(self::METHOD_POST, $route, $headers, $params, true, 30);
-
         $status = $response['headers']['status-code'];
+
         if ($status >= 400) {
-            throw new \Exception($response['body']['message'], $status);
-        } 
+            for ($attempts = 0; $attempts < 5; $attempts++) {
+                switch ($status) {
+                    case 404:
+                        $response = $this->createRuntime(
+                            functionId: $functionId,
+                            deploymentId: $deploymentId,
+                            projectId: $projectId,
+                            source: $path,
+                            runtime: $runtime,
+                            baseImage: $baseImage,
+                            vars: $vars,
+                            entrypoint: $entrypoint,
+                            commands: []
+                        );
+                        $response = $this->call(self::METHOD_POST, $route, $headers, $params, true, 30);
+                        $status = $response['headers']['status-code'];
+                        break;
+                    case 406:
+                        $response = $this->call(self::METHOD_POST, $route, $headers, $params, true, 30);
+                        $status = $response['headers']['status-code'];
+                        break;
+                    default:
+                        throw new \Exception($response['body']['message'], $status);
```

Comment:
This seems problematic to me. Doesn't this will throw error  even if status is say `200` (ie status is not 404 or 406) before reaching the line 157, because default behavior is to throw an exception?

Reply:
https://github.com/appwrite/appwrite/pull/2807/files#diff-23222bb861952e27ab08be6b8bbb449d43186436a2a110b5a24f3bbc61c287ffR157 

Since each switch case has a break statement, control comes out of the switch case. and this condition on like 157 ensures that we return the response body for any status code less than 400. 

I can explain further if its confusing. If you have pointers on how we could make it better, Id love to do that